### ISSUE TITLE
[Feature Request] Find a way to add a currency on a client's page and use that on an invoice

### ISSUE THREAD

**Issue by nielsdrost7** [InvoicePlane/InvoicePlane#941]
Description will follow

**Comment by theambidextrous** [comment]
Am not able to create a branch from this issue... permissions maybe?

**Comment by nielsdrost7** [comment]
@theambidextrous can you check again? I've given you more permissions

**Comment by theambidextrous** [comment]
hi... was caught up at work... let me check it out

**Comment by archergod** [comment]
+1 for this feature.

**Comment by Logicbloke** [comment]
A much needed feature, it feels like it's relatively simple to implement. It's just a visual thing, instead of grabbing it from the settings, grab it from the client's attributes. Am I understanding this correctly? 

I'll have to delve into the code myself but from a superficial point of view, it should be fairly straight-forward.

**Comment by nielsdrost7** [comment]
> I'll have to delve into the code myself but from a superficial point of view, it should be fairly straight-forward.

That's appreciated, man.

Normally the currency from the clients page also means showing that currency on the invoice and therefore having a difference between the seller's currency and the client's currency.

And that means ... exchange rates!
Ergo: **v2**

**But**:
- if you want to add a field in the database in the `ip_clients` table
- and you want to show it on `application/modules/clients/views/view.php`
- and you want to make it editable it on `application/modules/clients/views/form.php` (select box and let the Clients controller give the data to the view)
- and you want to make sure that `application/modules/clients/models/Mdl_client.php` deals with it (tiny bit of validation and save in the database
- and you want to show it on the Invoice PDF, Invoice "Web"
- and you probably want to show it on the guest pages / guest invoice views as well

Go for it.

If you want to develop, `dd()` and `dump()` are your friend.
Because you take the GitHub repository and you run `composer install` you'll get those "development tools" and you can do the `dd()` everywhere

**Comment by Logicbloke** [comment]
@nielsdrost7 Thanks for the detailed response and the invitation to contribute to this.

If I understood correctly, exchange rates are not available yet before v2? 

What about displaying sums and totals in both currencies in reports and such. It should avoid having to use any exchange rates, at least for v1.

I'll try to dedicate some time to start working on this.

**Comment by nielsdrost7** [comment]
> If I understood correctly, exchange rates are not available yet before v2?

Correct, it requires the baseprice and calculations based on exchange rates that change every day.

> What about displaying sums and totals in both currencies in reports and such. It should avoid having to use any exchange rates, at least for v1.

Great idea! This is the repository for InvoicePlane V2: https://github.com/InvoicePlane/InvoicePlane-v2

It's all divided in Modules, so I would go with the Core Module in this case.
For the Exchange Rates I would make calls to an Api that keep track of those exchange rates.

the baseprice is the price in the currency of the company and the calculated price is the price calculated with thd currency.

Most likely you'll also have to calculate the taxes that way as well.

If you want you can make unit tests, so it's easier to debug once bugs get reported in V2.

Thanks for contributing man!
Looking forward to seeing PR's come in for V2!
